api/interface.go

```diff
@@ -332,9 +332,19 @@ type Connection interface {
 	// will be easy to remove, but until we're using them via manifolds it's
 	// prohibitively ugly to do so.
 	Client() *Client
-	Uniter() (*uniter.State, error)
 	Upgrader() *upgrader.State
 	Reboot() (reboot.State, error)
 	InstancePoller() *instancepoller.API
 	UnitAssigner() unitassigner.API
 }
+
+// ConnectionUniter returns a version of the Connection that provides
+// functionality required by the uniter worker if possible else a non-nil error.
+func ConnectionUniter(c Connection) (*uniter.State, error) {
```

Comment:
Good to clean up the interface. This could probably go somewhere in api/agent/uniter. It looks like everyone who uses it already imports that package.

Also a note that it is only used in tests.

Reply:
Yeah I noticed it is only for tests.  I can try to move it to api/agent/uniter